src/eei.cpp

```diff
@@ -90,6 +90,78 @@ Literal EthereumInterface::callImport(Import *import, LiteralList& arguments) {
       return Literal();
     }
 
+    if (
+      import->base == Name("call") ||
+      import->base == Name("callCode") ||
+      import->base == Name("callDelegate") ||
+      import->base == Name("callStatic")
+    ) {
+      int64_t gas = arguments[0].geti64();
+      uint32_t addressOffset = arguments[1].geti32();
+      uint32_t valueOffset;
+      uint32_t dataOffset;
+      uint32_t dataLength;
+      uint32_t resultOffset;
+      uint32_t resultLength;
+
+      bool isValuePresent = import->base == Name("call") || import->base == Name("callCode");
+      evm_result call_result;
+      evm_message call_message;
+
+      if (isValuePresent) {
+      	valueOffset = arguments[2].geti32();
+	dataOffset = arguments[3].geti32();
+	dataLength = arguments[4].geti32();
+	resultOffset = arguments[5].geti32();
+	resultLength = arguments[6].geti32();
+
+	call_message.sender = msg.address;
+	call_message.value = loadUint128(valueOffset);
+	call_message.kind = (import->base == Name("callCode")) ? EVM_CALLCODE : EVM_CALL;
+	call_message.flags = 0;
```

Comment:
Apparently we need to copy the static flag. I suggest we do that outside this `if`:
```
heraAssert((msg.flags & ~EVM_STATIC) == 0, "Unknown flags not supported.");
call_message.flags = msg.flags;
```

And then do `call_message.flags |= EVM_STATIC` within the static call only.

Reply:
Check out this change in EVMJIT: https://github.com/ethereum/evmjit/commit/2f75569213e5710d096aec1976c1add2d0be3438#diff-5719df190800af8d47396050a508080e.